server/vulnerabilities/nvd/sync/cve_syncer.go

```diff
@@ -0,0 +1,727 @@
+package nvdsync
+
+import (
+	"bytes"
+	"compress/gzip"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"io/fs"
+	"net/http"
+	"os"
+	"path/filepath"
+	"sort"
+	"strconv"
+	"time"
+
+	"github.com/facebookincubator/nvdtools/cvefeed/nvd/schema"
+	"github.com/fleetdm/fleet/v4/orbit/pkg/constant"
+	"github.com/fleetdm/fleet/v4/pkg/fleethttp"
+	"github.com/fleetdm/fleet/v4/server/ptr"
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/pandatix/nvdapi/common"
+	"github.com/pandatix/nvdapi/v2"
+)
+
+// CVE syncs CVE information from the NVD database (nvd.nist.gov) using its API 2.0.
+// It stores the CVE information using the legacy feed format.
+type CVE struct {
+	client *http.Client
+	dbDir  string
+	logger log.Logger
+	debug  bool
+}
+
+var (
+	// timeBetweenRequests is the recommended time to wait between NVD API requests.
+	timeBetweenRequests = 6 * time.Second
+	// maxRetryAttempts is the maximum number of request to retry in case of API failure.
+	maxRetryAttempts = 10
+	// waitTimeForRetry is the time to wait between retries.
+	waitTimeForRetry = 30 * time.Second
+)
+
+// CVEOption allows configuring a CVE syncer.
+type CVEOption func(*CVE)
+
+// WithLogger sets the logger for a CVE syncer.
+//
+// Default value is log.NewNopLogger().
+func WithLogger(logger log.Logger) CVEOption {
+	return func(s *CVE) {
+		s.logger = logger
+	}
+}
+
+// WithDebug sets the debug mode for a CVE syncer.
+//
+// Default value is false.
+func WithDebug(debug bool) CVEOption {
```

Comment:
This doesn't appear to be used anywhere. Should it be?

Reply:
It was useful when debugging the API (in case we need to in the near future).